devlib/target.py

```diff
@@ -3138,6 +3138,416 @@ def _resolve_paths(self):
+        super(LocalTarget, self).__init__(connection_settings=connection_settings,
+                                          platform=platform,
+                                          working_directory=working_directory,
+                                          executables_directory=executables_directory,
+                                          connect=connect,
+                                          modules=modules,
+                                          load_default_modules=load_default_modules,
+                                          shell_prompt=shell_prompt,
+                                          conn_cls=conn_cls,
+                                          is_container=is_container,
+                                          max_async=max_async,
+                                          tmp_directory=tmp_directory,
+                                          )
+
+    @property
+    @memoized
+    def os(self):
+        """Detect the operating system"""
+        try:
+            uname_output = self.execute('uname -s').strip()
+            if uname_output == 'Darwin':
+                return 'darwin'
+            elif uname_output == 'Linux':
+                return 'linux'
+            else:
+                return 'unknown'
+        except TargetStableError:
+            return 'unknown'
+
+    def _is_macos(self):
+        """Check if running on macOS"""
+        return self.os == 'darwin'
+
+    def _is_linux(self):
+        """Check if running on Linux"""
+        return self.os == 'linux'
+
+    @property
+    @memoized
+    def abi(self):
+        """Get the system architecture"""
+        if self._is_macos():
+            try:
+                arch = self.execute('uname -m').strip()
+                # Map macOS architecture names to common names
+                arch_map = {
+                    'arm64': 'arm64',
+                    'x86_64': 'x86_64',
+                    'i386': 'x86',
+                }
+                return arch_map.get(arch, arch)
+            except TargetStableError:
+                return 'unknown'
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).abi
+
+    @property
+    @memoized
+    def os_version(self):
+        """Get OS version information"""
+        if self._is_macos():
+            try:
+                # Get macOS version
+                version = self.execute('sw_vers -productVersion').strip()
+                build = self.execute('sw_vers -buildVersion').strip()
+                name = self.execute('sw_vers -productName').strip()
+                return {
+                    'name': name,
+                    'version': version,
+                    'build': build,
+                }
+            except TargetStableError:
+                return {}
+        else:
+            # Use parent implementation for Linux and other systems
+            return super(LocalTarget, self).os_version
```

Comment:
`super(LocalTarget, self)` is just `super()` in this context. Arguments are only needed when:
1. The parser does not provide the interpreter enough context (e.g. in a nested function).
2. When trying to achieve something unusual (not following the normal MRO)
3. When trying to be compatible with Python 2 (official eol in 2020, devlib has dropped support for it a long time ago)